The job **`x86_64-gnu-aux`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8574848654/job/23502560336) [(plain)](https://github.com/rust-lang/rust/commit/ff3fccdff7ea329745b31afbb433344f514a6e7d/checks/23502560336/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
stderr:
error: unsupported operation: `getcwd` not available when isolation is enabled
##[error]   --> /checkout/library/std/src/sys/pal/unix/os.rs:161:17
    |
161 |             if !libc::getcwd(ptr, buf.capacity()).is_null() {
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `getcwd` not available when isolation is enabled
    = help: pass the flag `-Zmiri-disable-isolation` to disable isolation;
    = help: or pass `-Zmiri-isolation-error=warn` to configure Miri to return an error code from isolated operations (if supported for that operation) and continue with a warning
    = note: BACKTRACE:
    = note: BACKTRACE:
    = note: inside `std::sys::pal::unix::os::getcwd` at /checkout/library/std/src/sys/pal/unix/os.rs:161:17: 161:50
```

</details>

--- a/library/alloc/benches/lib.rs
+++ b/library/alloc/benches/lib.rs
@@ -1,6 +1,8 @@
 // Disabling on android for the time being
 // See https://github.com/rust-lang/rust/issues/73535#event-3477699747
 #![cfg(not(target_os = "android"))]
+// Disabling in Miri as these would take too long.
+#![cfg(not(miri))]
 #![feature(btree_extract_if)]
 #![feature(iter_next_chunk)]
 #![feature(repr_simd)]

--- a/library/core/benches/lib.rs
+++ b/library/core/benches/lib.rs
@@ -1,5 +1,7 @@
 // wasm32 does not support benches (no time).
 #![cfg(not(target_arch = "wasm32"))]
+// Disabling in Miri as these would take too long.
+#![cfg(not(miri))]
 #![feature(flt2dec)]
 #![feature(test)]
 #![feature(trusted_random_access)]

--- a/library/std/benches/lib.rs
+++ b/library/std/benches/lib.rs
@@ -1,3 +1,5 @@
+// Disabling in Miri as these would take too long.
+#![cfg(not(miri))]
 #![feature(test)]
 
 extern crate test;

--- a/library/std/tests/process_spawning.rs
+++ b/library/std/tests/process_spawning.rs
@@ -8,6 +8,7 @@ use std::str;
 mod common;
 
 #[test]
+#[cfg_attr(miri, ignore)] // Process spawning not supported by Miri
 fn issue_15149() {
     // If we're the parent, copy our own binary to a new directory.
     let my_path = env::current_exe().unwrap();

--- a/library/std/tests/switch-stdout.rs
+++ b/library/std/tests/switch-stdout.rs
@@ -51,6 +51,7 @@ fn switch_stdout_to(file: OwnedHandle) -> OwnedHandle {
 }
 
 #[test]
+#[cfg_attr(miri, ignore)] // dup/SetStdHandle not supported by Miri
 fn switch_stdout() {
     let temp = common::tmpdir();
     let path = temp.join("switch-stdout-output");

--- a/library/std/tests/thread.rs
+++ b/library/std/tests/thread.rs
@@ -5,7 +5,8 @@ use std::time::Duration;
 
 #[test]
 #[cfg_attr(target_os = "emscripten", ignore)]
-fn sleep() {
+#[cfg_attr(miri, ignore)] // Miri does not like the thread leak
+fn sleep_very_long() {
     let finished = Arc::new(Mutex::new(false));
     let t_finished = finished.clone();
     thread::spawn(move || {

--- a/src/bootstrap/mk/Makefile.in
+++ b/src/bootstrap/mk/Makefile.in
@@ -57,11 +57,21 @@ check-aux:
 		library/core \
 		library/alloc \
 		--no-doc
+	# Some doctests have intentional memory leaks.
 	$(Q)MIRIFLAGS="-Zmiri-ignore-leaks -Zmiri-disable-isolation" \
 		$(BOOTSTRAP) miri --stage 2 \
 		library/core \
 		library/alloc \
 		--doc
+	# In `std` we cannot test everything, so we test the most target-dependent modules.
+	$(Q)MIRIFLAGS="-Zmiri-disable-isolation" BOOTSTRAP_SKIP_TARGET_SANITY=1 \
+		$(BOOTSTRAP) miri --stage 2 library/std \
+		--no-doc -- \
+		--skip fs:: --skip net:: --skip process:: --skip sys::pal::
+	$(Q)MIRIFLAGS="-Zmiri-ignore-leaks -Zmiri-disable-isolation" BOOTSTRAP_SKIP_TARGET_SANITY=1 \
+		$(BOOTSTRAP) miri --stage 2 library/std \
+		--doc -- \
+		--skip fs:: --skip net:: --skip process:: --skip sys::pal::
 dist:
 	$(Q)$(BOOTSTRAP) dist $(BOOTSTRAP_ARGS)
 distcheck:

--- a/src/bootstrap/mk/Makefile.in
+++ b/src/bootstrap/mk/Makefile.in
@@ -52,6 +52,16 @@ check-aux:
 		src/tools/cargo \
 		src/tools/cargotest \
 		$(BOOTSTRAP_ARGS)
+	# Run standard library tests in Miri.
+	$(Q)$(BOOTSTRAP) miri --stage 2 \
+		library/core \
+		library/alloc \
+		--no-doc
+	$(Q)MIRIFLAGS="-Zmiri-ignore-leaks -Zmiri-disable-isolation" \
+		$(BOOTSTRAP) miri --stage 2 \
+		library/core \
+		library/alloc \
+		--doc
 dist:
 	$(Q)$(BOOTSTRAP) dist $(BOOTSTRAP_ARGS)
 distcheck:

